### ISSUE TITLE
Closure modules defined with `goog.provide` and `goog.require` do not work out-of-the-box, there is an error JSC_UNDEFINED_VARIABLE variable goog is undeclared

### ISSUE THREAD

**Issue by fingerartur** [google/closure-compiler#4048]
Closure modules defined with `goog.provide` and `goog.require` do not work out-of-the-box, there is an error during compilation.

Example:
```js
// ./src/js/index.js
goog.require('app.utils');

console.log('add 1,2', app.utils.add(1, 2));
```
```js
// ./src/js/utils.js
goog.provide('app.utils') // This raises ERROR - [JSC_UNDEFINED_VARIABLE] variable goog is undeclared

app.utils = {
  /**
   * @param {number} a 
   * @param {number} b 
   * @returns {number}
   */
  add: (a, b) => a + b,
}
```

**Calls to `goog.provide` and `goog.require` should not cause a compilation error.**

Additionally, the official wiki docs on Closure modules [here](https://github.com/google/closure-compiler/wiki/JS-Modules) and [here](https://github.com/google/closure-library/wiki/goog.module:-an-ES6-module-like-alternative-to-goog.provide) do not mention anything about this error or how to deal with it and neither does the FAQ. Ideally there would be a good straight-forward example of how to use Closure modules, which works out-of-the-box.

PS: As @lauraharker has pointed out, this issue can be hot-fixed by defining dummy typings, e.g. I can make the following changes to `index.js` in order to work around the error:
```js
// ./src/js/index.js
goog.require('app.utils');

console.log('add 1,2', app.utils.add(1, 2));

/**
* @type {{
*  require: Function,
*  provide: Function,
*  module: Function,
* }}
*/
var goog;
```

Related to #3506.

Compiler Version: v20221102

Build command:
```
java -jar ./scripts/closureCompiler.jar \
  --entry_point=./src/js/index.js \
  --js=./src/**.js \
  --dependency_mode=PRUNE \
  --warning_level=VERBOSE \
  --js_output_file=./dist/bundle.js \
  --module_resolution=WEBPACK \
  --compilation_level=ADVANCED \
  --jscomp_error=checkDebuggerStatement \
  --jscomp_error=unusedLocalVariables \
  --jscomp_error=reportUnknownTypes \
  --jscomp_error=strictCheckTypes;
```

**Comment by brad4d** [comment]
closure-compiler is intended to be used with closure-library.

I have added a note flagged as **IMPORTANT** here to point out that the `closure/goog/base.js` file should be included as a `--js` input to the compiler.

https://github.com/google/closure-library/wiki/goog.module:-an-ES6-module-like-alternative-to-goog.provide

**Comment by fingerartur** [comment]
I think it would be beneficial to add this documentation not only to the closure-library repo, but also to this (closure-compiler) repo. [Here is closure-compiler wiki article about available module systems](https://github.com/google/closure-compiler/wiki/JS-Modules) and base.js is not mentioned there once.
